src/workerd/api/node/buffer.c++

```diff
@@ -0,0 +1,396 @@
+// Copyright (c) 2017-2022 Cloudflare, Inc.
+// Licensed under the Apache 2.0 license found in the LICENSE file or at:
+//     https://opensource.org/licenses/Apache-2.0
+// Copyright Joyent and Node contributors. All rights reserved. MIT license.
+
+#include "buffer.h"
+#include <workerd/jsg/buffersource.h>
+#include <workerd/api/crypto-impl.h>
+#include <kj/encoding.h>
+#include <algorithm>
+
+// These are defined by <sys/byteorder.h> or <netinet/in.h> on some systems.
+// To avoid warnings, undefine them before redefining them.
+#ifdef BSWAP_2
+# undef BSWAP_2
+#endif
+#ifdef BSWAP_4
+# undef BSWAP_4
+#endif
+#ifdef BSWAP_8
+# undef BSWAP_8
+#endif
+
+#if defined(_MSC_VER)
+#include <intrin.h>
+#define BSWAP_2(x) _byteswap_ushort(x)
+#define BSWAP_4(x) _byteswap_ulong(x)
+#define BSWAP_8(x) _byteswap_uint64(x)
+#else
+#define BSWAP_2(x) ((x) << 8) | ((x) >> 8)
+#define BSWAP_4(x)                                                            \
```

Comment:
why note `__builtin_bswap32` et al.? Don't they work with clang?

Reply:
oh right, forgot about those ;-D

Reply:
That said, I don't believe all platforms have these yet, which is why node.js uses the macro definitions. I think just to be safe let's keep the macros.